### DISCUSSION TITLE
Why does formatOnType on for-loop affect other lines?

### DISCUSSION THREAD

**Discussion by ykadowak** [microsoft/vscode-cpptools#9983] [Q&A] [answered]
Let's say there is a code like this with `formatOnType` option and using `clang-format`
```cpp
for(int i = 0;) // <- the ";" triggers formatting


int foo = 0;
```

The result is
```cpp
for (int i = 0;) int foo = 0;
```

Here my question is why is the `int foo = 0` line affected by the formatting of the for loop line?

In other case, formatOnType only formats the line that is being typed. For example,
```cpp
int foo=0;
int bar=0; // <- Let's say this ";" is just typed
```

is formatted like below
```cpp
int foo=0; // <- this line is not affected
int bar = 0;
```

So why is the first example different?

**Comment by sean-mcmanus** [comment, marked as answer]
I'm not able to repro the bug. Can you provide more repro details? What clang-format version are you using? What other clang-format settings are you using? It may be a bug with clang-format itself.

If you set C_Cpp.loggingLevel to "Debug", the "C/C++" Output panel will show some formatting logging, i.e. inputs and raw outputs from clang-format which might help.

**Reply by ykadowak** [reply]
VSCode: Version: 1.72.0 (Universal)
Commit: 64bbfbf67ada9953918d72e1df2f4d8e537d340e
Date: 2022-10-04T23:21:58.256Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: No
Cpp-tools: v1.12.4

Using clang-format that is bundled with VSCode. `.clang-format` setting is `BasedOnType: Google` only.

**Repro process**:
> I edited the question because I forgot to put the close brace for the for-loop statement.
1. With the environment above(also `formatOnType` is on), prepare the code below. 
```cpp
for(int i = 0)


int foo = 0;
```

2. Type `;` at the end of the if statement
```cpp
for(int i = 0;)


int foo = 0;
```

3. Formats like below
```cpp
for(int i = 0;) int foo = 0;
```

**Log**
This is the log(somehow in Japanese even though my VSCode is in English 🤔 )
So it seems like VSCode is sending the ```int foo = 0;``` line as well. Is it by design or a bug? Also `FormatSelection` has the same behavior.

IMO, when you try to format the for-loop statement, it should not affect the lines that is more than one empty lines away.
```
cpptools/formatOnType: /Users/xxxx/main.cpp (id: 637)
入力の書式設定:
for (int i = 0;)


int foo = 0;

ドキュメントの書式設定: file:///Users/xxxx/main.cpp
Formatting Engine: clangFormat
未加工の出力の書式設定:
{ "Cursor": 15, "IncompleteFormat": false }
for (int i = 0;) int foo = 0;
```

**Reply by sean-mcmanus** [reply]
Thanks for the info. I repro the issue now. I've filed https://github.com/microsoft/vscode-cpptools/issues/10016 to track it.
